src/lib/environment.js

```diff
@@ -25,5 +30,16 @@ module.exports = {
   get instanceUrl() { return this.apiUrl && this.apiUrl.replace(/\/medic$/, ''); },
   get extraArgs() { return state.extraArgs; },
   get apiUrl() { return state.apiUrl; },
-  get force() { return state.force; }
+  get force() { return state.force; },
+
+  /**
+   * Return `true` if the environment seems to be production.
+   * @returns {boolean}
+   */
+  isProduction() {
+    if (!this.instanceUrl) return false;
+    const hostname = url.parse(this.instanceUrl).hostname;
+    if (LOCAL_MATCHER.test(hostname)) return false;
+    return !hostname.split('.').some(subdomain => DEV_MATCHER.test(subdomain));
```

Comment:
I think this is a little unreliable, for example, if the domain name just happens to start with `dev-`, however it's good enough for this issue.

Please raise a new issue to improve this function either by requiring users to set it explicitly, or querying to the server, or something.

Reply:
Yes those kind of addresses are taken into account, eg. executing the expression against a URL like you mention returns `false` as expected:

```
$ node
> const DEV_MATCHER = /^(dev|test|staging)(-\w+|\d+)?$/;
undefined
> let hostname = 'test-v2.other.mm.org';
undefined
> !hostname.split('.').some(subdomain => DEV_MATCHER.test(subdomain));
false
```

Reply:
Actually the last test was wrong, the intention in the title targeted cases like these but it wasn't the right URL to test, checkout the fix at https://github.com/medic/medic-conf/pull/342/commits/5db0a68d36e96ee4f832d313a43c860d20189283